### PR TITLE
chore: Add kb on migrating keyboard OSK to not use dotted cicrcle

### PR DESCRIPTION
**Pull request by darcywong00**
A quick attempt at documenting the migration steps @LornaSIL posted at
https://github.com/keymanapp/keyboards/issues/3112#issuecomment-2384164535

### CODE REVIEW

**Comment by mcdurdin** [comment]
I'm not sure this works as a KB. It'd be better to update the `&displayMap` reference at https://help.keyman.com/developer/language/reference/displaymap to discuss migration.

File: knowledge-base/kb0115.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+* xyz.kps
+
+    * add ../../../shared/fonts/kbd/kbdxxxx/KbdXxxx.ttf to list of fonts
+    * set **Keyboard font** to use the KbdXxxx font (used for OSK)
+
+* Needed only for the first keyboard using the Xxxx script: in release/shared/fonts/kbd/kbdxxxx/ folder, add KbdXxxx.json and KbdXxxx.ttf
+
+### Reference
+Please refer to [PR #9032](https://github.com/keymanapp/keyman/pull/9032) for more information
```
Comment (mcdurdin) [inline knowledge-base/kb0115.md:23]: Refer to displaymap docs, not PR

File: knowledge-base/kb0115.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# HOWTO: Migrate Keyboard OSK to use special OSK Font Instead of Dotted Circle (U+25CC)
+
+Some OSK's use a dotted circle (U+25CC) as the base for displaying a combining diacritic mark. This can lead to  inconsistent behavior when displayed across Operating Systems (double dotted circles, blank keys, etc). A prefered method is to update the keyboard to use a special OSK font for better display:
+
+* xyz.kmn
+
+    * Update version number;
+    * add store(&DISPLAYMAP) '../../..shared/fonts/kbd/kbdxxxx/KbdXxxx.json' (this file will be compiled into the .kmx so doesn't need including in the .kps)
```
Comment (mcdurdin) [inline knowledge-base/kb0115.md:8]:
```suggestion
    * add `store(&DISPLAYMAP) '../../..shared/fonts/kbd/kbdxxxx/KbdXxxx.json'` (this file will be compiled into the .kmx so doesn't need including in the .kps)
```

File: knowledge-base/kb0115.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    * set to use KbdXxxx font (in "encoding" "fontname" element).
+    * remove any U+25cc codes that were originally added to improve legibility
+
+* xyz.kps
+
+    * add ../../../shared/fonts/kbd/kbdxxxx/KbdXxxx.ttf to list of fonts
+    * set **Keyboard font** to use the KbdXxxx font (used for OSK)
+
+* Needed only for the first keyboard using the Xxxx script: in release/shared/fonts/kbd/kbdxxxx/ folder, add KbdXxxx.json and KbdXxxx.ttf
```
Comment (mcdurdin) [inline knowledge-base/kb0115.md:20]: Note that we should collect all the keyboards that use the same script and run the displaymap processing for them together; see displaymap docs for how to do  this.

Comment (darcywong00) [reply knowledge-base/kb0115.md:20]: From the repo there's:
* kbdhebr/KbdHebr.json
* kbdkmr/KbdKhmr.json

Comment (mcdurdin) [reply knowledge-base/kb0115.md:20]: Yeah, so far :grin:

File: knowledge-base/kb0115.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# HOWTO: Migrate Keyboard OSK to use special OSK Font Instead of Dotted Circle (U+25CC)
+
+Some OSK's use a dotted circle (U+25CC) as the base for displaying a combining diacritic mark. This can lead to  inconsistent behavior when displayed across Operating Systems (double dotted circles, blank keys, etc). A prefered method is to update the keyboard to use a special OSK font for better display:
+
+* xyz.kmn
+
+    * Update version number;
+    * add `store(&DISPLAYMAP) '../../..shared/fonts/kbd/kbdxxxx/KbdXxxx.json'` (this file will be compiled into the .kmx so doesn't need including in the .kps)
```
Comment (mcdurdin) [inline knowledge-base/kb0115.md:8]:
```suggestion
    * add `store(&DISPLAYMAP) '../../../shared/fonts/kbd/kbdxxxx/KbdXxxx.json'` (this file will be compiled into the .kmx so doesn't need including in the .kps)
```

File: knowledge-base/kb0115.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# HOWTO: Migrate Keyboard OSK to use special OSK Font Instead of Dotted Circle (U+25CC)
+
+Some OSK's use a dotted circle (U+25CC) as the base for displaying a combining diacritic mark. This can lead to  inconsistent behavior when displayed across Operating Systems (double dotted circles, blank keys, etc). A prefered method is to update the keyboard to use a special OSK font for better display:
```
Comment (mcdurdin) [inline knowledge-base/kb0115.md:3]: Can you use wrapped markdown, matching most of the rest of our content?

File: knowledge-base/kb0115.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+* xyz.kmn
+
+    * Update version number;
+    * add `store(&DISPLAYMAP) '../../..shared/fonts/kbd/kbdxxxx/KbdXxxx.json'` (this file will be compiled into the .kmx so doesn't need including in the .kps)
+
+* xyz.kvks
+
+    * set to use KbdXxxx font (in "encoding" "fontname" element).
```
Comment (mcdurdin) [inline knowledge-base/kb0115.md:12]: This is not necessary, and in fact we don't want to do this. This is a design-time and development-only font reference (or should be, see #9955 etc). The primary font info for deployment is in the .kps.

**Review by mcdurdin — APPROVED** [review]

File: knowledge-base/kb0115.md (hunk 6 above)
Comment (darcywong00) [reply knowledge-base/kb0115.md:12]:
```suggestion
```

**Pull request merged**